src/sql/workbench/contrib/commandLine/electron-sandbox/commandLine.ts

```diff
@@ -118,37 +118,27 @@ export class CommandLineWorkbenchContribution implements IWorkbenchContribution,
 			});
 			return;
 		}
-		let connectedContext: azdata.ConnectedContext = undefined;
-		if (profile) {
-			if (profile.providerName && !this._capabilitiesService.providers[profile.providerName]) {
-				const installed = await this._connectionManagementService.handleUnsupportedProvider(profile.providerName);
-				if (!installed) {
-					// User cancelled install prompt so exit early since we won't be able to connect
-					return;
+		if (commandName) {
+			if (profile && commandName === Command.connect) {
```

Comment:
Maybe this logic should be reversed, with the connect handler only handling the connect-specific logic, and have the handleURL function go through processCommandLine directly instead (which then calls the appropriate command handler). It just seems like it'd be cleaner to have the command-specific logic each in their own handler.